### PR TITLE
Cleanup step

### PR DESCRIPTION
**Pull request by alexreinking**
Replace the cleanup-artifacts script that recently stopped working (missing `tar.gz` extension) with a custom buildbot step and remove it from the repo. Also remove `worker_heartbeat.py`, which is unused. Already tested working on beta-master.

### CODE REVIEW

**Comment by steven-johnson** [comment]
...I gotta say that the new code to clean up stuff is waaaaay less obvious than the old code.

**Comment by abadams** [comment]
I think it just needs some comments. Can you imagine trying to grok the old version without the comment that's as long as the code?

**Comment by alexreinking** [comment]
> ...I gotta say that the new code to clean up stuff is waaaaay less obvious than the old code.

I added some comments and made things a bit simpler (YAGNI). Still it does a little more than the old code, namely reports removals and failures on a per-file basis.

File: master/master.cfg [hunk 1]
```diff
@@ -888,7 +888,9 @@ def add_halide_cmake_package_steps(factory, builder_type):
                    mode=0o644,
                    masterdest=get_distrib_name.withArgs(version, target, ext)))
 
-    def pkg_version_and_target(_, path: Path):
+    def pkg_version_and_target(path: Path):
+        # Archives names are formatted like: Halide-[verison]-[arch]-[commit].[ext]
```
Comment (steven-johnson) [inline master/master.cfg:892]: nit: s/verison/version/

**Review by steven-johnson — APPROVED** [review]

File: master/master.cfg (hunk 1 above)
Comment (alexreinking) [reply master/master.cfg]: Oops. Thanks for the catch.

**Pull request merged**